Fix/47/Update-R504-for-assignment-value

CalumY opened the pull request:
This PR updates the `visit_Assign` function to get `refs` for the assignment value. This fixes a couple of the issues that can be found in #47, b0109af5fe27986d5372a54a2cfd43d1fff97efd. All other issues can be refactored to get R504 to pass (see 47c5b3c64c832acc9ba679fe50ab1b12b5df821b). 

As @ericbn, [notes](https://github.com/afonasev/flake8-return/issues/47#issuecomment-1077714924), this rule could be considered controversial, so it could potentially be updated to be optional in another issue/pr.  

Fixes #47.

afonasev commented:
Thx for contribute! Can you fix https://github.com/afonasev/flake8-return/actions/runs/3041122888/jobs/4900937759?

CalumY commented:
@afonasev, I have updated the linting errors and fixed some other issues identified in the actions. This can be seen [here](https://github.com/CalumY/flake8-return/actions/runs/3043505721/jobs/4902847896).

Please let me know if you require any other updates.
